spec/models/program_spec.rb

```diff
@@ -240,6 +240,19 @@
     end
   end
 
+  describe '#any_event_for_this_date?' do
+    let(:event){ create(:event, program: program) }
+    let!(:event_schedule) { create(:event_schedule, event: event) }
+
+    it 'returns true if there is any event for particular date' do
+      expect(program.any_event_for_this_date?(DateTime.now.to_date)).to eq true
```

Comment:
It is not clear to me why the event is not scheduled on this particular date, because I don't know when the event is supposed to be scheduled. I would make this clear in the test, ie add a specific date in let definition.